File: app/src/main/java/cn/intret/app/picgo/model/image/MoveFileDetectResult.java
package cn.intret.app.picgo.model.image;

import android.util.Pair;

import java.io.File;
import java.util.List;


public class MoveFileDetectResult {

    private File mTargetDir;
    private List<Pair<File, File>> mCanMoveFiles;
    private List<Pair<File, File>> mConflictFiles;

    public File getTargetDir() {
        return mTargetDir;
    }

    public MoveFileDetectResult setTargetDir(File targetDir) {
        mTargetDir = targetDir;
        return this;
    }

    public MoveFileDetectResult setCanMoveFiles(List<Pair<File, File>> canMoveFiles) {
        mCanMoveFiles = canMoveFiles;
        return this;
    }

    public List<Pair<File, File>> getCanMoveFiles() {
        return mCanMoveFiles;
    }

    public MoveFileDetectResult setConflictFiles(List<Pair<File, File>> conflictFiles) {
        mConflictFiles = conflictFiles;
        return this;
    }

    public List<Pair<File, File>> getConflictFiles() {
        return mConflictFiles;
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/image/data/MoveFileDetectResult.kt
package cn.intret.app.picgo.model.image.data

import android.util.Pair

import java.io.File


class MoveFileDetectResult {

    private var targetDir: File? = null
    internal var canMoveFiles: List<Pair<File, File>>? = null
    var conflictFiles: List<Pair<File, File>>? = null

    fun getTargetDir(): File? {
        return targetDir
    }

    fun setTargetDir(targetDir: File): MoveFileDetectResult {
        this.targetDir = targetDir
        return this
    }

    fun setCanMoveFiles(canMoveFiles: List<Pair<File, File>>): MoveFileDetectResult {
        this.canMoveFiles = canMoveFiles
        return this
    }

    fun getCanMoveFiles(): List<Pair<File, File>>? {
        return canMoveFiles
    }

    fun setConflictFiles(conflictFiles: List<Pair<File, File>>): MoveFileDetectResult {
        this.conflictFiles = conflictFiles
        return this
    }
}
